apps/api/src/app/workflows-v2/util/template-variables/extract-template-variables.ts

```diff
@@ -1,37 +1,79 @@
 import { Template, Liquid } from 'liquidjs';
 
 /**
- * Extracts variable names from a Liquid template string by parsing and traversing the tokens.
- * Returns a Set of dot-notation variable paths found in the template.
- *
- * @param str - The Liquid template string to parse
- * @returns Array of variable paths found in the template
- * @example
- * const template = "Hello {{user.name}}, your order #{{order.id}} is {{order.status}}";
- * extractTemplateVars(template)
- * // Returns Array [ "user.name", "order.id", "order.status" ]
+ * Extracts variable names from a Liquid template
  */
-export const extractTemplateVars = function (str: string): string[] {
-  if (str === null || str === undefined || typeof str !== 'string') {
-    return [];
+export const extractTemplateVars = function (str: string): {
+  validVariables: string[];
+  invalidVariables: { context: string; message: string; variable: string }[];
+} {
+  if (str === null || str === undefined || str === '' || typeof str !== 'string') {
+    return { validVariables: [], invalidVariables: [] };
+  }
+
+  const liquidExpressions = extractLiquidExpressions(str);
+  if (liquidExpressions.length === 0) {
+    return { validVariables: [], invalidVariables: [] };
+  }
+
+  let variables = new Set<string>();
+  const invalidVariables: { context: string; message: string; variable: string }[] = [];
+
+  for (const liquidExpression of liquidExpressions) {
+    try {
+      variables = new Set([...variables, ...parseByLiquid(liquidExpression)]);
+    } catch (error) {
+      invalidVariables.push(
+        ...error.errors.map((e) => ({ context: e.context, message: e.message, variable: liquidExpression }))
```

Comment:
This object is not 100% correct because here i am passing liquid js Output and not a variable.
Id suggest keeping it like this because there is no valid(transitivity) way to know what the variable is in an invalid Output.
wanted to hear your thoughts on this.

```suggestion
        ...error.errors.map((e) => ({ context: e.context, message: e.message, output: liquidExpression }))
```